From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 Nov 2017 19:03:42 +0100
Subject: [PATCH] {BUG in 4.15-rc1} drivers/soc/mediatek: initialize rdata
 before testing write-read

clang reports an uninitialize variable:

    drivers/soc/mediatek/mtk-pmic-wrap.c:1127:6: error: variable 'rdata'
    is used uninitialized whenever '||' condition is true
    [-Werror,-Wsometimes-uninitialized]
            if (pwrap_write(wrp, wrp->slave->dew_regs[PWRAP_DEW_WRITE_TEST],
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/soc/mediatek/mtk-pmic-wrap.c:1132:39: note: uninitialized
    use occurs here
                    dev_err(wrp->dev, "rdata=0x%04X\n", rdata);
                                                        ^~~~~
    drivers/soc/mediatek/mtk-pmic-wrap.c:1127:6: note: remove the '||'
    if its condition is always false
            if (pwrap_write(wrp, wrp->slave->dew_regs[PWRAP_DEW_WRITE_TEST],
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/soc/mediatek/mtk-pmic-wrap.c:1057:11: note: initialize the
    variable 'rdata' to silence this warning
            u32 rdata;
                     ^
                      = 0
    1 error generated.
---
 drivers/soc/mediatek/mtk-pmic-wrap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/mediatek/mtk-pmic-wrap.c b/drivers/soc/mediatek/mtk-pmic-wrap.c
index 2afae64061d8..ef731d5f4fb5 100644
--- a/drivers/soc/mediatek/mtk-pmic-wrap.c
+++ b/drivers/soc/mediatek/mtk-pmic-wrap.c
@@ -1124,6 +1124,7 @@ static int pwrap_init_cipher(struct pmic_wrapper *wrp)
 	pwrap_writel(wrp, 1, PWRAP_CIPHER_MODE);
 
 	/* Write Test */
+	rdata = 0;
 	if (pwrap_write(wrp, wrp->slave->dew_regs[PWRAP_DEW_WRITE_TEST],
 			PWRAP_DEW_WRITE_TEST_VAL) ||
 	    pwrap_read(wrp, wrp->slave->dew_regs[PWRAP_DEW_WRITE_TEST],
-- 
